experimental/oak_baremetal_kernel/src/mm/mod.rs

```diff
@@ -65,5 +68,44 @@ pub fn init<const N: usize, E: E820Entry>(
         })
         .for_each(|range| alloc.mark_valid(range, true));
 
+    // Step 2: mark known in-use regions as not available.
+
+    // First, leave out the first 2 MiB as there be dragons (and bootloader data structures)
+    alloc.mark_valid(
+        PhysFrame::range_inclusive(
+            PhysFrame::from_start_address(PhysAddr::new(0x0)).unwrap(),
+            PhysFrame::from_start_address(PhysAddr::new(0x0)).unwrap(),
+        ),
+        false,
+    );
+
+    // Second, mark every `PT_LOAD` section from the phdrs as used.
+    program_headers
+        .iter()
+        .filter(|phdr| phdr.p_type == PT_LOAD)
```

Comment:
What about the stack and the rest of the BSS entries? I don't think they will be PT_LOAD headers?

Reply:
They do.

```
Program Headers:
  Type           Offset   VirtAddr           PhysAddr           FileSiz  MemSiz   Flg Align
  LOAD           0x000000 0x0000000000200000 0x0000000000200000 0x000190 0x000190 R   0x1000
  LOAD           0x001000 0x0000000000201000 0x0000000000201000 0x000048 0x000048 R E 0x1000
  LOAD           0x002000 0xffffffff80400000 0x0000000000400000 0x25e2ab 0x25e2ab R E 0x1000
  LOAD           0x261000 0xffffffff80800000 0x0000000000800000 0x049cb0 0x049cb0 R   0x1000
  LOAD           0x2ab000 0xffffffff80a00000 0x0000000000a00000 0x000068 0x000068 RW  0x1000
  LOAD           0x400000 0xffffffff80c00000 0x0000000000c00000 0x000000 0x280000 RW  0x200000

 Section to Segment mapping:
  Segment Sections...
   00     
   01     .boot 
   02     .text 
   03     .rodata 
   04     .data .got 
   05     .bss .stack
```
Note that `.bss` and `.stack`, both marked `(NOLOAD)` in the linker script and placed in a segment with `FileSiz` of `0x000000` (indicating there's nothing to load from the file), the segment still has the type `LOAD`.